src/main/java/org/broadinstitute/hellbender/tools/walkers/variantutils/ReblockGVCF.java

```diff
@@ -0,0 +1,453 @@
+ * gatk ReblockGVCF \
+ *   -R reference.fasta \
+ *   --variant sample1.g.vcf \
+ *
+ *   -O sample1.reblocked.g.vcf
+ * </pre>
+ *
+ * <h3>Caveats</h3>
+ * <p>Only single-sample gVCF files produced by HaplotypeCaller can be used as input for this tool.</p>
+ * <p>By default this tool only passes through annotations used by VQSR.  A different set of annotations can be specified with the usual -A argument.  Annotation groups are not supported by this tool.</p>
+ *
+ * <h3>Special note on ploidy</h3>
+ * <p>This tool assumes diploid genotypes.</p>
+ *
+ */
+@ExperimentalFeature
+@CommandLineProgramProperties(summary = "Compress a single-sample GVCF from HaplotypeCaller by merging homRef blocks using new GQ band parameters",
+        oneLineSummary = "Condenses homRef blocks in a single-sample GVCF",
+        programGroup = OtherProgramGroup.class,
+        omitFromCommandLine = true)
+@DocumentedFeature
+public class ReblockGVCF extends VariantWalker {
+
+    private static String GVCF_BLOCK = "GVCFBlock";
+    private static int PLOIDY_TWO = 2;  //FIXME: hack to avoid extra arg collections
+
+    private VariantContextWriter vcfWriter;
+
+    @Argument(fullName = StandardArgumentDefinitions.OUTPUT_LONG_NAME, shortName = StandardArgumentDefinitions.OUTPUT_SHORT_NAME,
+            doc="File to which variants should be written")
+    private File outputFile;
+
+    @Argument(fullName="includeNonVariantSites", shortName="allSites", doc="Include loci found to be non-variant after genotyping")
+    public boolean INCLUDE_NON_VARIANTS = true;
+
+    @ArgumentCollection
+    public GenotypeCalculationArgumentCollection genotypeArgs = new GenotypeCalculationArgumentCollection();
+
+    /**
+     * Which annotations to copy over for the reblocked output gVCF file.
+     */
+    @Advanced
+    @Argument(fullName="annotation", shortName="A", doc="One or more specific annotations to recompute.  The single value 'none' removes the default annotations")
+    protected List<String> annotationsToUse = Arrays.asList(VCFConstants.DEPTH_KEY, VCFConstants.RMS_MAPPING_QUALITY_KEY, GATKVCFConstants.READ_POS_RANK_SUM_KEY, GATKVCFConstants.MAP_QUAL_RANK_SUM_KEY, GATKVCFConstants.RAW_RMS_MAPPING_QUALITY_KEY);
+
+    @Advanced
+    @Argument(fullName="GVCFGQBands", shortName="GQB", doc="Exclusive upper bounds for reference confidence GQ bands " +
+            "(must be in [1, 100] and specified in increasing order)")
+    public List<Integer> GVCFGQBands = new ArrayList<>();
+    {
+        GVCFGQBands.add(20); GVCFGQBands.add(100);
+    };
+
+    @Advanced
+    @Argument(fullName="drop-low-qual-variants", shortName="drop-low-quals", doc="Exclude variants and homRef blocks that are GQ0 from the reblocked GVCF to save space")
+    protected boolean dropLowQuals = false;
+
+    @Advanced
+    @Argument(fullName="RGQ-threshold-to-drop", shortName="RGQ-threshold", doc="Reference genotype quality (PL[0]) value below which variant sites will be converted to GQ0 homRef calls")
```

Comment:
Should this have a min value of 0.0?

Reply:
I tried to sneak this one past because I don't like the new style.

The default is zero so that the default output only has homRef blocks modified, but Eric was convinced we could get better compression by throwing out the low quality variants, which is probably true but we didn't do a real test.  The arg name could be more accurate.